cirq/ops/common_gates.py

```diff
@@ -377,7 +377,7 @@ def __repr__(self) -> str:
         ).format(proper_repr(self._exponent), self._global_shift)
 
 
-@value.value_equality
```

Comment:
so the issue here is that this override eigen_gate's decorator?  Do we need to do this for XPowGate and YPowGate?

Reply:
Actually.. turns out the test I added passed anyways. Removed this change.

Reply:
OK, I think there is still a bug here from one of the cases I found, but will follow up separately.